-1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-2f9ce7/32960/summary.html
**COMMIT**: 79dd00584883af0ebfe5a27e9aff49a1ba8ad67c
**CMSSW**: CMSSW_13_2_X_2023-06-03-1100/el8_amd64_gcc11
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-externals/llvm-project/12/32960/install.sh` to create a dev area with all the needed externals and cmssw changes.


## External Build

I found compilation error when building: 

<pre>[3736/6082] /data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/el8_amd64_gcc11/external/gcc/11.2.1-f9b9dfdd886f71cd63f5538223d8f161/bin/c++ -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/BUILD/el8_amd64_gcc11/external/llvm/16.0.3-e26122adce53be773bc8b56dd72b2010/build/tools/clang/lib/CodeGen -I/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/BUILD/el8_amd64_gcc11/external/llvm/16.0.3-e26122adce53be773bc8b56dd72b2010/llvm-16.0.3-bd1b99af77c08abf7a677e2ae688e57afd0f922f/clang/lib/CodeGen -I/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/BUILD/el8_amd64_gcc11/external/llvm/16.0.3-e26122adce53be773bc8b56dd72b2010/llvm-16.0.3-bd1b99af77c08abf7a677e2ae688e57afd0f922f/clang/include -I/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/BUILD/el8_amd64_gcc11/external/llvm/16.0.3-e26122adce53be773bc8b56dd72b2010/build/tools/clang/include -I/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/BUILD/el8_amd64_gcc11/external/llvm/16.0.3-e26122adce53be773bc8b56dd72b2010/build/include -I/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/BUILD/el8_amd64_gcc11/external/llvm/16.0.3-e26122adce53be773bc8b56dd72b2010/llvm-16.0.3-bd1b99af77c08abf7a677e2ae688e57afd0f922f/llvm/include -fPIC -fno-semantic-interposition -fvisibility-inlines-hidden -Werror=date-time -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wno-missing-field-initializers -pedantic -Wno-long-long -Wimplicit-fallthrough -Wno-maybe-uninitialized -Wno-class-memaccess -Wno-redundant-move -Wno-pessimizing-move -Wno-noexcept-type -Wdelete-non-virtual-dtor -Wsuggest-override -Wno-comment -Wno-misleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -ffunction-sections -fdata-sections -fno-common -Woverloaded-virtual -fno-strict-aliasing -O3 -DNDEBUG -std=c++17 -MD -MT tools/clang/lib/CodeGen/CMakeFiles/obj.clangCodeGen.dir/TargetInfo.cpp.o -MF tools/clang/lib/CodeGen/CMakeFiles/obj.clangCodeGen.dir/TargetInfo.cpp.o.d -o tools/clang/lib/CodeGen/CMakeFiles/obj.clangCodeGen.dir/TargetInfo.cpp.o -c /data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/BUILD/el8_amd64_gcc11/external/llvm/16.0.3-e26122adce53be773bc8b56dd72b2010/llvm-16.0.3-bd1b99af77c08abf7a677e2ae688e57afd0f922f/clang/lib/CodeGen/TargetInfo.cpp
[3737/6082] /data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/el8_amd64_gcc11/external/gcc/11.2.1-f9b9dfdd886f71cd63f5538223d8f161/bin/c++ -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/BUILD/el8_amd64_gcc11/external/llvm/16.0.3-e26122adce53be773bc8b56dd72b2010/build/tools/clang/lib/CodeGen -I/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/BUILD/el8_amd64_gcc11/external/llvm/16.0.3-e26122adce53be773bc8b56dd72b2010/llvm-16.0.3-bd1b99af77c08abf7a677e2ae688e57afd0f922f/clang/lib/CodeGen -I/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/BUILD/el8_amd64_gcc11/external/llvm/16.0.3-e26122adce53be773bc8b56dd72b2010/llvm-16.0.3-bd1b99af77c08abf7a677e2ae688e57afd0f922f/clang/include -I/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/BUILD/el8_amd64_gcc11/external/llvm/16.0.3-e26122adce53be773bc8b56dd72b2010/build/tools/clang/include -I/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/BUILD/el8_amd64_gcc11/external/llvm/16.0.3-e26122adce53be773bc8b56dd72b2010/build/include -I/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/BUILD/el8_amd64_gcc11/external/llvm/16.0.3-e26122adce53be773bc8b56dd72b2010/llvm-16.0.3-bd1b99af77c08abf7a677e2ae688e57afd0f922f/llvm/include -fPIC -fno-semantic-interposition -fvisibility-inlines-hidden -Werror=date-time -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wno-missing-field-initializers -pedantic -Wno-long-long -Wimplicit-fallthrough -Wno-maybe-uninitialized -Wno-class-memaccess -Wno-redundant-move -Wno-pessimizing-move -Wno-noexcept-type -Wdelete-non-virtual-dtor -Wsuggest-override -Wno-comment -Wno-misleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -ffunction-sections -fdata-sections -fno-common -Woverloaded-virtual -fno-strict-aliasing -O3 -DNDEBUG -std=c++17 -MD -MT tools/clang/lib/CodeGen/CMakeFiles/obj.clangCodeGen.dir/CodeGenModule.cpp.o -MF tools/clang/lib/CodeGen/CMakeFiles/obj.clangCodeGen.dir/CodeGenModule.cpp.o.d -o tools/clang/lib/CodeGen/CMakeFiles/obj.clangCodeGen.dir/CodeGenModule.cpp.o -c /data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/BUILD/el8_amd64_gcc11/external/llvm/16.0.3-e26122adce53be773bc8b56dd72b2010/llvm-16.0.3-bd1b99af77c08abf7a677e2ae688e57afd0f922f/clang/lib/CodeGen/CodeGenModule.cpp
[3738/6082] /data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/el8_amd64_gcc11/external/gcc/11.2.1-f9b9dfdd886f71cd63f5538223d8f161/bin/c++ -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/BUILD/el8_amd64_gcc11/external/llvm/16.0.3-e26122adce53be773bc8b56dd72b2010/build/tools/clang/lib/Analysis -I/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/BUILD/el8_amd64_gcc11/external/llvm/16.0.3-e26122adce53be773bc8b56dd72b2010/llvm-16.0.3-bd1b99af77c08abf7a677e2ae688e57afd0f922f/clang/lib/Analysis -I/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/BUILD/el8_amd64_gcc11/external/llvm/16.0.3-e26122adce53be773bc8b56dd72b2010/llvm-16.0.3-bd1b99af77c08abf7a677e2ae688e57afd0f922f/clang/include -I/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/BUILD/el8_amd64_gcc11/external/llvm/16.0.3-e26122adce53be773bc8b56dd72b2010/build/tools/clang/include -I/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/BUILD/el8_amd64_gcc11/external/llvm/16.0.3-e26122adce53be773bc8b56dd72b2010/build/include -I/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/BUILD/el8_amd64_gcc11/external/llvm/16.0.3-e26122adce53be773bc8b56dd72b2010/llvm-16.0.3-bd1b99af77c08abf7a677e2ae688e57afd0f922f/llvm/include -fPIC -fno-semantic-interposition -fvisibility-inlines-hidden -Werror=date-time -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wno-missing-field-initializers -pedantic -Wno-long-long -Wimplicit-fallthrough -Wno-maybe-uninitialized -Wno-class-memaccess -Wno-redundant-move -Wno-pessimizing-move -Wno-noexcept-type -Wdelete-non-virtual-dtor -Wsuggest-override -Wno-comment -Wno-misleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -ffunction-sections -fdata-sections -fno-common -Woverloaded-virtual -fno-strict-aliasing -O3 -DNDEBUG -std=c++17 -MD -MT tools/clang/lib/Analysis/CMakeFiles/obj.clangAnalysis.dir/ExprMutationAnalyzer.cpp.o -MF tools/clang/lib/Analysis/CMakeFiles/obj.clangAnalysis.dir/ExprMutationAnalyzer.cpp.o.d -o tools/clang/lib/Analysis/CMakeFiles/obj.clangAnalysis.dir/ExprMutationAnalyzer.cpp.o -c /data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/BUILD/el8_amd64_gcc11/external/llvm/16.0.3-e26122adce53be773bc8b56dd72b2010/llvm-16.0.3-bd1b99af77c08abf7a677e2ae688e57afd0f922f/clang/lib/Analysis/ExprMutationAnalyzer.cpp
[3739/6082] /data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/el8_amd64_gcc11/external/gcc/11.2.1-f9b9dfdd886f71cd63f5538223d8f161/bin/c++ -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/BUILD/el8_amd64_gcc11/external/llvm/16.0.3-e26122adce53be773bc8b56dd72b2010/build/tools/clang/lib/CodeGen -I/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/BUILD/el8_amd64_gcc11/external/llvm/16.0.3-e26122adce53be773bc8b56dd72b2010/llvm-16.0.3-bd1b99af77c08abf7a677e2ae688e57afd0f922f/clang/lib/CodeGen -I/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/BUILD/el8_amd64_gcc11/external/llvm/16.0.3-e26122adce53be773bc8b56dd72b2010/llvm-16.0.3-bd1b99af77c08abf7a677e2ae688e57afd0f922f/clang/include -I/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/BUILD/el8_amd64_gcc11/external/llvm/16.0.3-e26122adce53be773bc8b56dd72b2010/build/tools/clang/include -I/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/BUILD/el8_amd64_gcc11/external/llvm/16.0.3-e26122adce53be773bc8b56dd72b2010/build/include -I/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/BUILD/el8_amd64_gcc11/external/llvm/16.0.3-e26122adce53be773bc8b56dd72b2010/llvm-16.0.3-bd1b99af77c08abf7a677e2ae688e57afd0f922f/llvm/include -fPIC -fno-semantic-interposition -fvisibility-inlines-hidden -Werror=date-time -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wno-missing-field-initializers -pedantic -Wno-long-long -Wimplicit-fallthrough -Wno-maybe-uninitialized -Wno-class-memaccess -Wno-redundant-move -Wno-pessimizing-move -Wno-noexcept-type -Wdelete-non-virtual-dtor -Wsuggest-override -Wno-comment -Wno-misleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -ffunction-sections -fdata-sections -fno-common -Woverloaded-virtual -fno-strict-aliasing -O3 -DNDEBUG -std=c++17 -MD -MT tools/clang/lib/CodeGen/CMakeFiles/obj.clangCodeGen.dir/CGBuiltin.cpp.o -MF tools/clang/lib/CodeGen/CMakeFiles/obj.clangCodeGen.dir/CGBuiltin.cpp.o.d -o tools/clang/lib/CodeGen/CMakeFiles/obj.clangCodeGen.dir/CGBuiltin.cpp.o -c /data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/BUILD/el8_amd64_gcc11/external/llvm/16.0.3-e26122adce53be773bc8b56dd72b2010/llvm-16.0.3-bd1b99af77c08abf7a677e2ae688e57afd0f922f/clang/lib/CodeGen/CGBuiltin.cpp
ninja: build stopped: subcommand failed.
error: Bad exit status from /data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/tmp/rpm-tmp.6HFQBQ (%build)


RPM build errors:
line 37: It's not recommended to have unversioned Obsoletes: Obsoletes: external+llvm+16.0.3-e26122adce53be773bc8b56dd72b2010
Macro expanded in comment on line 350: %{pkginstroot}/lib64
</pre>

--- a/clang/lib/Analysis/IssueHash.cpp
+++ b/clang/lib/Analysis/IssueHash.cpp
@@ -98,6 +98,8 @@ static std::string GetEnclosingDeclContextSignature(const Decl *D) {
     case Decl::Record:
     case Decl::CXXRecord:
     case Decl::Enum:
+    case Decl::Field:
+    case Decl::Var:
       DeclName = ND->getQualifiedNameAsString();
       break;
     case Decl::CXXConstructor:
@@ -110,9 +112,6 @@ static std::string GetEnclosingDeclContextSignature(const Decl *D) {
     case Decl::ObjCMethod:
       // ObjC Methods can not be overloaded, qualified name uniquely identifies
       // the method.
-      DeclName = ND->getQualifiedNameAsString();
-      break;
-    case Decl::FieldDecl:
       DeclName = ND->getQualifiedNameAsString();
       break;
     default:

